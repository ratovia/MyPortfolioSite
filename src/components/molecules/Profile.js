import React from "react"
import styled from "styled-components"
import { CircleImageIcon } from "../atoms/CircleImageIcon"
import { TypographyH6 } from "../atoms/Typography"

const ProfileArea = styled.div`
  display: flex;
  flex-direction: column;
  align-item: center;
  max-width: 365px;
`

const ProfileItem = styled.div`
  text-align: left;
  margin-bottom: 15px;
`

const ProfileImage = styled.div`
  text-align: center;
  margin-bottom: 72px;
`

export const Profile = ({ profileText, skillText, imageUrl }) => {
  return (
    <ProfileArea>
      <ProfileImage>
        <CircleImageIcon imageUrl={imageUrl} size="large"></CircleImageIcon>
      </ProfileImage>
      <ProfileItem>
        <TypographyH6 label={profileText} color="white"></TypographyH6>
      </ProfileItem>
      <ProfileItem>
        <TypographyH6 label={skillText} color="white"></TypographyH6>
      </ProfileItem>
    </ProfileArea>
  )
}
