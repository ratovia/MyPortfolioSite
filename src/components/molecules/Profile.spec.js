import React from "react"
import renderer from "react-test-renderer"
import { Profile } from "./Profile"
import javascriptIcon from "../../stories/assets/javascript_icon.png"

describe("Profile", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Profile
          profileText="Hello world"
          skillText="This is Component"
          imageUrl={javascriptIcon}
        />
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
