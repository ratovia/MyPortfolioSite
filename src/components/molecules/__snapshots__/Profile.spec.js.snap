// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile renders correctly 1`] = `
<div
  className="sc-jRQAMF jFTfeO"
>
  <div
    className="sc-iCfLBT eyXoBR"
  >
    <img
      alt="アイコン"
      className="sc-bdvvaa hVPPTo"
      height="240px"
      src="test-file-stub"
      width="240px"
    />
  </div>
  <div
    className="sc-gKckTs epwxeJ"
  >
    <h6
      className="sc-eCImvq kPbozt"
    >
      <p
        className="sc-gsDJrp dOmOMC"
        color="white"
      >
        Hello world
      </p>
    </h6>
  </div>
  <div
    className="sc-gKckTs epwxeJ"
  >
    <h6
      className="sc-eCImvq kPbozt"
    >
      <p
        className="sc-gsDJrp dOmOMC"
        color="white"
      >
        This is Component
      </p>
    </h6>
  </div>
</div>
`;
